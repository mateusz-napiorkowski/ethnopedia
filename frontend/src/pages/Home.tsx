import React from "react";
import { useUser } from "../providers/UserProvider";
import CollectionsPage from "./collections/CollectionsPage";
import LandingPage from "./LandingPage";

const Home = () => {
    const { jwtToken } = useUser();
    return jwtToken ? <CollectionsPage /> : <LandingPage />;
};

export default Home;
