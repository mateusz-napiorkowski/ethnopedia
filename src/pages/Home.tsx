import Table from "../components/Table"
import Navbar from "../components/Navbar"
import Sidebar from "../components/Sidebar"
import { useEffect, useState } from "react"

const Home = () => {

    return <div className="flex flex-col">
        <div className="flex flex-row flex-grow">
            <Sidebar />
            <div className="flex flex-grow flex-col">
                <Navbar />
                <Table />
            </div>
        </div>
    </div>
}

export default Home
