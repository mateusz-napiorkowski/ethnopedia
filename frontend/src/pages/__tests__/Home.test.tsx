import '@testing-library/jest-dom';
import { render } from '@testing-library/react'
import Home from "../Home"
import { MemoryRouter, Routes, Route } from "react-router-dom";
import { QueryClient, QueryClientProvider } from "react-query";

const queryClient = new QueryClient();

const renderComponent = () => {
    return render(
        <QueryClientProvider client={queryClient}>
            <MemoryRouter initialEntries={[`/`]}>
                <Routes>
                    <Route path="/" element={<Home/>}/>
                </Routes>  
            </MemoryRouter>
        </QueryClientProvider>
    );
};

describe("Home tests", () => {
    beforeEach(() => {
        queryClient.clear();
        jest.clearAllMocks()
    });

    it("should render home page (navbar and collections page in loading state)", () => {           
        const {container} = renderComponent()
        
        expect(container).toMatchSnapshot()
    })
})