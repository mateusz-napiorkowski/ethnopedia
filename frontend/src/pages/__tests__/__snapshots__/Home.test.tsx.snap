// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tests should render home page (navbar and collections page in loading state) 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <div
      class="flex-grow"
    >
      <nav
        class="bg-gray-800 border-gray-200 dark:bg-gray-200 w-full border-b dark:border-gray-600"
      >
        <div
          class="max-w-screen-xl flex flex-wrap justify-between mx-auto p-4"
        >
          <div
            class="flex items-center mt-1"
          >
            <h1
              class="logo text-gray-200 dark:text-gray-900 text-6xl "
            >
              ethnopedia
            </h1>
          </div>
          <div
            class="flex items-center md:order-2"
          >
            <button
              class="text-gray-800 bg-white border border-gray-300 focus:outline-none
                           hover:bg-gray-100 focus:ring-4 focus:ring-gray-200 font-medium rounded-lg text-sm
                           p-2.5 mr-2 dark:bg-gray-800 dark:text-white dark:border-gray-600
                           dark:hover:bg-gray-700 dark:hover:border-gray-600 dark:focus:ring-gray-700"
              type="button"
            >
              <svg
                class="w-4 h-4 m-0.5"
              />
            </button>
            <button
              class="mr-2 bg-blue-500 hover:bg-blue-400 font-semibold text-white border-none"
              type="button"
            >
              Zaloguj się
            </button>
            <button
              class="bg-blue-500 hover:bg-blue-400 font-semibold text-white border-none"
              type="button"
            >
              Zarejestruj się
            </button>
          </div>
        </div>
      </nav>
      <div
        class="flex justify-center items-center h-screen bg-gray-50 dark:bg-gray-900"
        role="status"
      >
        <svg />
      </div>
    </div>
  </div>
</div>
`;
